src/main/java/com/rackspace/ceres/app/services/QueryService.java

```diff
@@ -58,27 +60,27 @@ public QueryService(ReactiveCqlTemplate cqlTemplate,
   }
 
   public Flux<QueryResult> queryRaw(String tenant, String metricName,
-                                    Map<String, String> queryTags,
-                                    Instant start, Instant end) {
-    // given the queryTags filter, locate the series-set that apply
+      Map<String, String> queryTags,
+      Instant start, Instant end) {
     return metadataService.locateSeriesSetHashes(tenant, metricName, queryTags)
         // then perform a retrieval for each series-set
-        .flatMap(seriesSet ->
-            mapSeriesSetResult(tenant, seriesSet,
-                // over each time slot partition of the [start,end) range
-                Flux.fromIterable(timeSlotPartitioner
-                    .partitionsOverRange(start, end, null)
-                )
-                    .concatMap(timeSlot ->
-                        cqlTemplate.queryForRows(
-                            dataTablesStatements.rawQuery(),
-                            tenant, timeSlot, seriesSet, start, end
-                        )
-                            .name("queryRaw")
-                            .metrics()
-                    )
-            )
-        )
+        .flatMap(seriesSet -> {
+          Metadata metadata = buildMetaData(Aggregator.raw, start, end, null);
+          return mapSeriesSetResult(tenant, seriesSet,
+              // over each time slot partition of the [start,end) range
+              Flux.fromIterable(timeSlotPartitioner
+                  .partitionsOverRange(start, end, null)
+              )
+                  .concatMap(timeSlot ->
+                      cqlTemplate.queryForRows(
+                          dataTablesStatements.rawQuery(),
+                          tenant, timeSlot, seriesSet, start, end
+                      )
+                          .name("queryRaw")
+                          .metrics()
+                  ), metadata
+          );
+        })
```

Comment:
There's an artform to keeping diffs minimal and also maintaining the "chaining" look of functional programming. This should "diff" better:

```suggestion
        .flatMap(seriesSet -> 
          mapSeriesSetResult(tenant, seriesSet,
              // over each time slot partition of the [start,end) range
              Flux.fromIterable(timeSlotPartitioner
                  .partitionsOverRange(start, end, null)
              )
                  .concatMap(timeSlot ->
                      cqlTemplate.queryForRows(
                          dataTablesStatements.rawQuery(),
                          tenant, timeSlot, seriesSet, start, end
                      )
                          .name("queryRaw")
                          .metrics()
                  ), buildMetaData(Aggregator.raw, start, end, null);
          )
        )
```